### PR TITLE
More stylistic updates to the manual for consistency

### PR DESCRIPTION
**Pull request by ararslan**
In the same vein as #21456, my goal is to make the stylistic conventions in the manual consistent, e.g. by migrating to `where` and `struct` syntax, using the standard 4 spaces for indentation, etc.

CI skipped, remove from commit message when merging.

### CODE REVIEW

File: doc/src/manual/arrays.md [hunk 1]
```diff
@@ -716,7 +716,7 @@ Julia sparse matrices have the type `SparseMatrixCSC{Tv,Ti}`, where `Tv` is the
 values, and `Ti` is the integer type for storing column pointers and row indices.:
 
 ```julia
-type SparseMatrixCSC{Tv,Ti<:Integer} <: AbstractSparseMatrix{Tv,Ti}
+mutable struct SparseMatrixCSC{Tv,Ti<:Integer} <: AbstractSparseMatrix{Tv,Ti}
```
Comment (tkelman) [inline doc/src/manual/arrays.md]: this one's actually not mutable any more

File: doc/src/manual/faq.md [hunk 2]
```diff
@@ -29,7 +29,7 @@ you can redefine types and constants.  You can't import the type names into `Mai
 to be able to redefine them there, but you can use the module name to resolve the scope.  In other
 words, while developing you might use a workflow something like this:
 
-```
+```julia
```
Comment (StefanKarpinski) [inline doc/src/manual/faq.md]: Why is this necessary? Isn't this the default?

Comment (ararslan) [inline doc/src/manual/faq.md]: Nope. The default is no syntax coloring. Adding `julia` highlights Julia keywords, etc.

Comment (StefanKarpinski) [inline doc/src/manual/faq.md]: Is that true in Documenter?

Comment (ararslan) [inline doc/src/manual/faq.md]: Yes. If you look at this rendered in the docs as they stand (see https://docs.julialang.org/en/latest/manual/faq.html), the block has no color.

Comment (StefanKarpinski) [inline doc/src/manual/faq.md]: Shouldn't we change the default for Documenter code blocks to Julia code instead of putting `julia` on blocks everywhere?

Comment (ararslan) [inline doc/src/manual/faq.md]: Not necessarily--how else do you get a plain text block?

Comment (StefanKarpinski) [inline doc/src/manual/faq.md]: Write ```` ```text ````?

Comment (ararslan) [inline doc/src/manual/faq.md]: Hm, didn't know you could do that. Then yeah, that's something to consider. In the meantime is this alright?

Comment (tkelman) [inline doc/src/manual/faq.md]: guess that would be equivalent to treating docstrings as a julia specific markdown extension, which i guess we already do with even numbers of backticks using math mode. worth an issue at documenter for discussion maybe.

Comment (ararslan) [inline doc/src/manual/faq.md]: That's a good idea. Regarding this PR, is there anything that should be changed, added, removed, etc, or is it okay as-is?

Comment (tkelman) [inline doc/src/manual/faq.md]: lgtm, but no rush unless your next one will conflict. others may notice something I didn't​

Comment (ararslan) [inline doc/src/manual/faq.md]: This PR is the last for the manual. Might go through the devdocs or something next, but nothing will conflict. I don't mean to rush, I just want to try to direct discussion back to the matter at hand. 😉

**Review by StefanKarpinski — APPROVED** [review]

File: doc/src/manual/performance-tips.md [hunk 3]
```diff
@@ -389,17 +389,17 @@ Note the somewhat surprising fact that `T` doesn't appear in the declaration of
 that we'll return to in a moment. With this approach, one can write functions such as:
 
 ```jldoctest containers2
-julia> function myfunc{T<:Integer, A<:AbstractArray}(c::MyContainer{T,A})
+julia> function myfunc(c::MyContainer{T,A}) where {T<:Integer, A<:AbstractArray}
```
Comment (nalimilan) [inline doc/src/manual/performance-tips.md]: Why not do `MyContainer{<:Integer, <:AbstractArray}`?

Comment (ararslan) [inline doc/src/manual/performance-tips.md]: Good idea, I'll go with that. Thanks!
